pkg/controllers/openshift-route-controller-manager.go

```diff
@@ -56,44 +58,37 @@ func (s *OCPRouteControllerManager) Dependencies() []string {
 
 func (s *OCPRouteControllerManager) configure(cfg *config.MicroshiftConfig) {
 	s.kubeconfig = cfg.KubeConfigPath(config.KubeAdmin)
-
-	if err := s.writeConfig(cfg); err != nil {
-		klog.Fatalf("Failed to write route-controller-manager config %v", err)
-	}
-
-	var configFilePath = cfg.DataDir + "/resources/route-controller-manager/config/config.yaml"
-	s.ConfigFilePath = configFilePath
-	s.Output = os.Stdout
+	s.config = s.writeConfig(cfg)
 }
 
-func (s *OCPRouteControllerManager) writeConfig(cfg *config.MicroshiftConfig) error {
+func (s *OCPRouteControllerManager) writeConfig(cfg *config.MicroshiftConfig) *openshiftcontrolplanev1.OpenShiftControllerManagerConfig {
 	servingCertDir := cryptomaterial.RouteControllerManagerServingCertDir(cryptomaterial.CertsDirectory(cfg.DataDir))
 
-	// OCM config contains a list of controllers to enable/disable.
-	// If no list is specified, all controllers are started (default).
-	// If a non-zero length list is specified, only controllers enabled in the list are started.  Unlisted controllers
-	// are therefore disabled.  Enable controllers by appending their name to `controllers:`. Disable a controller by
-	// prepending "-" to the name, e.g. `controllers: ["-openshift.io/build"]
-	// Disabled OCM controllers are included in the list for documentary purposes.
-	data := []byte(`apiVersion: openshiftcontrolplane.config.openshift.io/v1
-kind: OpenShiftControllerManagerConfig
-kubeClientConfig:
-  kubeConfig: ` + s.kubeconfig + `
-  connectionOverrides:
-    contentType: "application/json"
-servingInfo:
-  bindAddress: "0.0.0.0:8445"
-  certFile: ` + cryptomaterial.ServingCertPath(servingCertDir) + `
-  keyFile:  ` + cryptomaterial.ServingKeyPath(servingCertDir) + `
-  clientCA: ` + cryptomaterial.TotalClientCABundlePath(cryptomaterial.CertsDirectory(cfg.DataDir)) + `
-controllers:
-- "openshift.io/ingress-ip"
-- "openshift.io/ingress-to-route"
-`)
-
-	path := filepath.Join(cfg.DataDir, "resources", "route-controller-manager", "config", "config.yaml")
-	os.MkdirAll(filepath.Dir(path), os.FileMode(0700))
-	return ioutil.WriteFile(path, data, 0644)
+	c := &openshiftcontrolplanev1.OpenShiftControllerManagerConfig{
+		KubeClientConfig: configv1.KubeClientConfig{
+			KubeConfig: s.kubeconfig,
```

Comment:
Could you please add the ContentType param of ConnectionOverrides like this:

https://github.com/openshift/microshift/pull/988/commits/c630434974e3f61bb5ab5938e292ba0ed68417f3

We would want to avoid issues in case protobuf is the default format for communication.

Reply:
done, and also set `BindNetwork` to  `tcp4`